### PR TITLE
Introduce make lint

### PR DESCRIPTION
**Pull request by schuellerf**
trying to reproduce `golangci-lint` errors with the new `make lint` and fix them

### CODE REVIEW

File: tools/install_golangci_lint_deps.sh
Comment (achilleas-k) [inline tools/install_golangci_lint_deps.sh:1]: These aren't `golangci-lint` dependencies, they're osbuild-composer (and obsuild-worker) deps.  I'd actually prefer to emphasise the fact that it's a script that uses `apt` and ubuntu package names and name it something like `apt-install-deps.sh`, so people don't try to use it to install dependencies in other scenarios.

File: tools/install_golangci_lint_deps.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# just a warning - as "sudo" is not in the container we are using
+echo "This script should be run as root"
+
+apt-get update
+
+# This is needed to lint internal/upload/koji package
+apt-get install -y libkrb5-dev
```
Comment (mvo5) [inline tools/install_golangci_lint_deps.sh:9]: (quick drive-by comment) doing this in one long command line (like `apt install -y libkrb5-dev libgpgme-dev libbtrfs-dev libdevmapper-dev` is slightly more efficient as it will read/rebuild the apt cache only once (but of course writing comments is a little bit harden then.

File: Makefile [hunk 2]
```diff
@@ -275,3 +284,11 @@ scratch: $(RPM_SPECFILE) $(RPM_TARBALL)
...
 		$(RPM_SPECFILE)
 
+container_composer_golangci_built.info: Makefile Containerfile_golangci_lint tools/install_golangci_lint_deps.sh
+	podman build -f Containerfile_golangci_lint -t $(GOLANGCI_COMPOSER_IMAGE) --build-arg "GOLANGCI_LINT_VERSION=$(GOLANGCI_LINT_VERSION)"
+	echo "Image last built on" > $@
+	date >> $@
+
+.PHONY: lint
+lint: $(GOLANGCI_LINT_CACHE_DIR) container_composer_golangci_built.info
+	podman run -t --rm -v $(SRCDIR):/app:z -v $(GOLANGCI_LINT_CACHE_DIR):/root/.cache:z -w /app $(GOLANGCI_COMPOSER_IMAGE) golangci-lint run -v
```
Comment (achilleas-k) [inline Makefile:294]: I'm not entirely sure what benefit this has over just installing (or downloading) the golangci-lint version we care about and running it without a container.  I'm not against it, but I personally will never run this instead of `go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.53 run`.

File: tools/install_golangci_lint_deps.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
```
Comment (mvo5) [inline tools/install_golangci_lint_deps.sh:1]: All the other tools use the explicit `#!/bin/bash` here it seems, unless there is a good reason I would keep it this way. I personally prefer the `#!/bin/sh`  (or `#!/bin/bash`) because it ensures it's always the distro provided bash [0] that is used, not some random one installed in $PATH. But then I'm coming from a distro background so take it with a grain of salt :)

[0] assuming no "make install prefix=/" was used

**Review by mvo5 — COMMENTED** [review]
Some quick dry-by comments (that may actually be a bit silly, freel free to ignore)

File: Makefile (hunk 2 above)
Comment (schuellerf) [reply Makefile:294]: The environment would be more similar to the GH workflow, which is the one rejecting your PR later on 🤔

File: tools/install_golangci_lint_deps.sh (hunk 1 above)
Comment (schuellerf) [reply tools/install_golangci_lint_deps.sh:9]: yes, I just took it over (was like this) due to the comments - I also didn't really like calling apt that often

File: tools/install_golangci_lint_deps.sh (hunk 3 above)
Comment (schuellerf) [reply tools/install_golangci_lint_deps.sh:1]: I thought `/usr/bin/env` is the most generic way but the main reason was, that my IDE suggested and autocompleted that line 🙊

Comment (schuellerf) [reply tools/install_golangci_lint_deps.sh:1]: I'd rather have a small chat in the team about the way to go and I'd be happy to align all then.

Comment (schuellerf) [reply tools/install_golangci_lint_deps.sh:1]: you are right about the `sh` though - `bash` is not needed in this script ;)

File: Makefile (hunk 2 above)
Comment (achilleas-k) [reply Makefile]: I guess that's fair.

**Comment by achilleas-k** [comment]
Fedora 38 RPM builds are failing but F38 is EOL so let's just drop them.

**Comment by schuellerf** [comment]
how to re-run the gitlab pipelines? I think the PR should be fine

**Review by achilleas-k — APPROVED** [review]
